src/Features/CSharp/Portable/Braces/CSharpBracesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,404 @@
+            _infoRemoveBracesDiagnosticDescriptor = CreateRemoveBracesDiagnosticDescriptor(DiagnosticSeverity.Info);
+            _warningRemoveBracesDiagnosticDescriptor = CreateRemoveBracesDiagnosticDescriptor(DiagnosticSeverity.Warning);
+            _errorRemoveBracesDiagnosticDescriptor = CreateRemoveBracesDiagnosticDescriptor(DiagnosticSeverity.Error);
+
+            _severityToRemoveBracesDescriptorMap =
+                new Dictionary<DiagnosticSeverity, DiagnosticDescriptor>
+                {
+                    {DiagnosticSeverity.Hidden, _noneRemoveBracesDiagnosticDescriptor },
+                    {DiagnosticSeverity.Info, _infoRemoveBracesDiagnosticDescriptor },
+                    {DiagnosticSeverity.Warning, _warningRemoveBracesDiagnosticDescriptor },
+                    {DiagnosticSeverity.Error, _errorRemoveBracesDiagnosticDescriptor },
+                };
+        }
+
+        private DiagnosticDescriptor CreateAddBracesDiagnosticDescriptor(DiagnosticSeverity severity) =>
+            new DiagnosticDescriptor(
+                IDEDiagnosticIds.AddBracesDiagnosticId,
+                s_addBracesLocalizableTitle,
+                s_addBracesLocalizableMessage,
+                DiagnosticCategory.Style,
+                severity,
+                isEnabledByDefault: true);
+
+
+        private DiagnosticDescriptor CreateRemoveBracesDiagnosticDescriptor(DiagnosticSeverity severity) =>
+            new DiagnosticDescriptor(
+                IDEDiagnosticIds.RemoveBracesDiagnosticId,
+                s_removeBracesLocalizableTitle,
+                s_removeBracesLocalizableMessage,
+                DiagnosticCategory.Style,
+                severity,
+                isEnabledByDefault: true);
+
+        public bool RunInProcess => false;
+
+        public override void Initialize(AnalysisContext context)
+        {
+            var syntaxKindsOfInterest =
+            ImmutableArray.Create(SyntaxKind.IfStatement,
+                SyntaxKind.ElseClause,
+                SyntaxKind.ForStatement,
+                SyntaxKind.ForEachStatement,
+                SyntaxKind.ForEachComponentStatement,
+                SyntaxKind.WhileStatement,
+                SyntaxKind.DoStatement,
+                SyntaxKind.UsingStatement,
+                SyntaxKind.LockStatement,
+                SyntaxKind.FixedStatement);
+
+            context.RegisterSyntaxNodeAction(AnalyzeNode, syntaxKindsOfInterest);
+        }
+
+        public DiagnosticAnalyzerCategory GetAnalyzerCategory() => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        public void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var optionSet = context.Options.GetOptionSet();
+            var useBracesOption = optionSet.GetOption(CSharpCodeStyleOptions.UseBracesWherePossible);
+            var severity = useBracesOption.Notification.Value;
+
+            if (useBracesOption.Value)
```

Comment:
something here feels weird.  It seems like the options are such taht we'll always report errors, even for users who don't care one way or another.  i.e. you can only say "you must use braces" or "you must not use braces".  What about users who want to say "i don't care"?

@dpoeschl ?